storage/innobase/xtrabackup/src/backup_copy.cc

```diff
@@ -644,8 +646,14 @@ bool write_ibd_buffer(ds_file_t *file, unsigned char *buf, size_t buf_len,
           compressed_len = ut_calc_align(compressed_len, block_size);
         }
         sparse_map.push_back(ds_sparse_chunk_t{skip, len + compressed_len});
-        skip = page_size - compressed_len;
-        len = 0;
+        if (i != total_pages) {
```

Comment:
won't
```
      if ((Compression::is_compressed_page(page) ||
          fil_page_get_type(page) == FIL_PAGE_COMPRESSED_AND_ENCRYPTED) && !last_page_of_file) {
```
be better condition?

Also, the last page in the buffer is not always the last page of file. This patch will write non-sparse page every 10M.